sdk/src/main/java/io/horizen/utils/BytesUtils.java

```diff
@@ -231,13 +235,13 @@ public static byte[] fromHorizenMcTransparentAddress(String address, NetworkPara
         byte[] prefix = Arrays.copyOfRange(addressBytes, 0, HORIZEN_ADDRESS_PREFIX_LENGTH);
         if(params instanceof MainNetParams) {
             if(!Arrays.equals(prefix, PUBLIC_KEY_MAINNET_PREFIX) && !Arrays.equals(prefix, PUBLIC_KEY_MAINNET_PREFIX_OLD) &&
-                    !Arrays.equals(prefix, SCRIPT_MAINNET_PREFIX) && !Arrays.equals(prefix, SCRIPT_MAINNET_PREFIX_OLD))
+                    (onlyPubKey || !Arrays.equals(prefix, SCRIPT_MAINNET_PREFIX) && !Arrays.equals(prefix, SCRIPT_MAINNET_PREFIX_OLD)))
                 throw new IllegalArgumentException(String.format(
                         "Incorrect Horizen address format, pubKey or script MainNet prefix expected, got %s",
```

Comment:
error message must be different if onlyPubKey = true:
"pubKey or script" => "pubKety"
(also below)

Reply:
Done